skopt/optimizer/optimizer.py

```diff
@@ -393,11 +403,6 @@ def tell(self, x, y, fit=True):
 
         * `y` [scalar or list]:
             Value of objective at `x`.
-
-        * `fit` [bool, default=True]
```

Comment:
we need to keep the `fit` argument with a deprecation warning for two releases. I would also like to keep it because I do use it as a way of adding observations to the optimizer when I know I have a batch of them and don't want to re-run the fitting because I will add several before calling `ask` again.

Reply:
All the fit logic is moved to ask, so the fit argument for tell won't do anything.

Reply:
Ah sorry. I thought I posted a comment (somewhere :-/) about the reasoning why the work of `fit` is done in `tell`.

TL;DR: we should keep it in `tell`. My reasoning is the following: the cycle of ask-and-tell is always: ask, evaluate expensive function, tell, check if we are done, if not ask, eval, tell, check if done, ...

For example when I `tell` a point I just evaluated I expect the surrogate I get from the optimizer to include information from that point so that I can decide if I need to ask for another point or if we are done or what else to do. Similarly if we were to plot the surrogate model after one round (ask, eval, tell) we'd expect it to be fitted to that one point. It feels a bit weird to me having to do: ask, eval, tell, ask, now plot and decide what we should do next.

What do you think?